database-plugins/src/main/java/io/cdap/plugin/db/common/FQNGenerator.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.db.common;
+
+import io.cdap.plugin.common.db.DBUtils;
+import io.opentracing.contrib.jdbc.ConnectionInfo;
+import io.opentracing.contrib.jdbc.parser.URLParser;
+
+/**
+ * Generate FQN from DB URL Connection
+ */
+
+public class FQNGenerator {
+
+  public static String constructFQN(String url, String tableName) {
+    // dbtype, host, port, db from the connection string
+    // table is the reference name
+    ConnectionInfo connectionInfo = URLParser.parse(url);
+    // DB type as set by library after extraction
+    if (connectionInfo.getDbType() == DBUtils.POSTGRESQL_TAG) {
```

Comment:
How is this working for postgres?
See https://stackoverflow.com/questions/513832/how-do-i-compare-strings-in-java

Reply:
Checked in the debugger, it is evaluating to true with ==. Attached screenshot

![image](https://user-images.githubusercontent.com/88705714/194226477-c6722b14-9eae-4b83-b30d-34b515cb98e8.png)